odoo/http.py

```diff
@@ -1561,6 +1562,49 @@ def _response(self, result=None, error=None):
             ('Content-Length', len(body)),
         ])
 
+class JsonDataDispatcher(Dispatcher):
+    routing_type = 'jsondata'
+
+    @classmethod
+    def is_compatible_with(cls, request):
+        return request.httprequest.mimetype == 'application/json'
```

Comment:
This PR is in draft and I'm not planning on actually reviewing it (only testing) but should this be
```suggestion
        return request.httprequest.mimetype == JSON_MIMETYPE
```
or even better: `in JSON_MIMETYPES`

Reply:
+1 for using the constante. IMO when you explicitely set application/json-rpc, you should go to type='json', not type='jsondata'